aurora/model/decoder.py

```diff
@@ -71,14 +72,23 @@ def __init__(
                 separate Perceiver.
             modulation_head (bool, optional): Enable an additional head, the so-called modulation
                 head, that can be used to predict the difference. Defaults to `False`.
+            predict_difference_history_dim_lookup (dict[str, int]): For every variable that we want
+                to predict the difference for, the index into the history dimension that should be
+                used when predicting the difference.
         """
         super().__init__()
 
         # If additional modulation heads are required, simulate them as different variables with
         # the suffix `_mod`.
-        if modulation_head:
-            surf_vars += tuple(f"{name}_mod" for name in surf_vars)
-            atmos_vars += tuple(f"{name}_mod" for name in atmos_vars)
+        if modulation_head and predict_difference_history_dim_lookup:
+            surf_vars += tuple(
+                f"{name}_mod"
+                for name in list(set(surf_vars) & set(predict_difference_history_dim_lookup))
```

Comment:
```suggestion
                for name in surf_vars if name in predict_difference_history_dim_lookup
```

Could we perhaps do it this way and also below? This would leave the order of `surf_vars` unchanged and not rely on the order that comes out of a `list(set(...))`.